kitsune/messages/api.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+
+from django.contrib.auth.models import Group, User
+from django.db.models import Q
+from django.views.decorators.http import require_GET
+
+from kitsune.access.decorators import login_required
+from kitsune.sumo.decorators import json_view
+from kitsune.sumo.utils import webpack_static
+from kitsune.users.templatetags.jinja_helpers import profile_avatar
+
+
+@login_required
+@require_GET
+@json_view
+def get_autocomplete_suggestions(request):
+    """An API to provide auto-complete data for user names or groups."""
+    pre = request.GET.get("term", "") or request.GET.get("query", "")
+    if not pre or not request.user.is_authenticated:
+        return []
+
+    def get_suggestions(pre, is_staff=False):
+        """Generate autocomplete suggestions for users and, if applicable, groups."""
+
+        def create_suggestion(item, item_type, is_user=True):
+            """Create a dictionary object for the autocomplete suggestion."""
+            return {
+                "type": item_type,
```

Comment:
Nits.
- Instead of passing in `item_type`, you could use something like `item.__class__.__name__` or `"User" if isinstance(item, User) else "Group"`.
- Instead of passing in `is_user`, you could use `isinstance(item, User)`.

Reply:
Got it.